### PR TITLE
Sb separate use from creation

### PR DESCRIPTION
**Pull request by stevebillings**
Refactored PolarisBuildStep and ExecutePolarisCliStep so their run() methods only do: (a) validation, (b) create the step factory and workflow object to delegate to, and (c) call the workflow object to do the real work.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStep.java [hunk 1]
```diff
@@ -236,7 +129,14 @@ public boolean isApplicable(final Class<? extends AbstractProject> jobType) {
         public String getDisplayName() {
             return DISPLAY_NAME;
         }
-
     }
 
+    private void validateBuild(final AbstractBuild<?, ?> build) throws AbortException {
```
Comment (rottebds) [inline src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStep.java]: I've always been under the impression that all logic a class uses should come before any inner class declarations. Is there a compelling reason to leave validateBuild() after the DescriptorImpl class definition?

Comment (rottebds) [reply src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStep.java]: Sidebar: should validation be moved to the Workflow classes?

File: src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStepWorkflow.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jenkins.extensions.ChangeBuildStatusTo;
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.stepworkflow.StepWorkflow;
+import com.synopsys.integration.stepworkflow.StepWorkflowResponse;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Result;
+
+public class PolarisBuildStepWorkflow {
```
Comment (rottebds) [inline src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStepWorkflow.java:39]: There seems to be an implied pattern here-- would it serve us to establish a contract for these StepWorkflow building classes, or would that simply be architecture for the sake of architecture?

File: src/test/java/com/synopsys/integration/jenkins/polaris/extensions/pipeline/ExecutePolarisCliStepTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+@PowerMockIgnore({"javax.crypto.*", "javax.net.ssl.*" })
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({PolarisCli.class, StepWorkflow.class})
+public class ExecutePolarisCliStepTest {
+    private static final String WORKSPACE_REL_PATH = "out/test/PolarisBuildStepTest/testPerform/workspace";
+    private static final String TEST_POLARIS_CLI_NAME = "testPolarisCliName";
+    private static final String TEST_POLARIS_HOME = "/tmp/polaris";
+    public static final String TEST_POLARIS_ARGS = "testArgs";
+
+    // TODO: To improve this test (to test more of ExecutePolarisCliStep.Execution.run(), we could refactor ExecutePolarisCliStep to:
```
Comment (rottebds) [inline src/test/java/com/synopsys/integration/jenkins/polaris/extensions/pipeline/ExecutePolarisCliStepTest.java:37]: Is this TODO done?

File: src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStep.java (hunk 1 above)
Comment (stevebillings) [reply src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStep.java]: I agree on both counts. My next commit will move validate() to the Workflow class.

File: src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStepWorkflow.java (hunk 2 above)
Comment (stevebillings) [reply src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStepWorkflow.java:39]: I wondered the same thing. I got hung up on the different returns types of the perform() method. But maybe there's a good option I haven't thought of.

File: src/test/java/com/synopsys/integration/jenkins/polaris/extensions/pipeline/ExecutePolarisCliStepTest.java (hunk 3 above)
Comment (stevebillings) [reply src/test/java/com/synopsys/integration/jenkins/polaris/extensions/pipeline/ExecutePolarisCliStepTest.java:37]: Not yet, but the two test TODOs are basically next. I think these changes will help.

**Pull request merged**